Simplify how experimental PromQL functions are enabled (#10660)
grafana/mimir: CHANGELOG.md

* [CHANGE] Ingester: Set `-ingester.ooo-native-histograms-ingestion-enabled` to true by default. #10483
* [CHANGE] Ruler: Add `user` and `reason` labels to `cortex_ruler_write_requests_failed_total` and `cortex_ruler_queries_failed_total`; add `user` to
    `cortex_ruler_write_requests_total` and `cortex_ruler_queries_total` metrics. #10536

> * [CHANGE] Querier / Query-frontend: Remove experimental `-querier.promql-experimental-functions-enabled` and `-query-frontend.block-promql-experimental-functions` CLI flag (and respective YAML config option) to enable experimental PromQL functions. Instead access to experimental PromQL functions is always blocked and can be selectively enabled by the per-tenant setting `enabled_promql_experimental_functions`. #10660

Reviewer: ```suggestion
* [CHANGE] Querier / Query-frontend: Remove experimental `-querier.promql-experimental-functions-enabled` and `-query-frontend.block-promql-experimental-functions` CLI flags and respective YAML configuration options to enable experimental PromQL functions. Instead access to experimental PromQL functions is always blocked. You can enable them using the per-tenant setting `enabled_promql_experimental_functions`. #10660
```
